ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx

```diff
@@ -0,0 +1,220 @@
+
+import NotFoundMessage from 'Components/NotFoundMessage/NotFoundMessage';
+import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate/EmptyStateTemplate';
+import CheckboxSelect from 'Components/PatternFly/CheckboxSelect';
+import ReportParametersDetails from '../components/ReportParametersDetails';
+import DeliveryDestinationsDetails from '../components/DeliveryDestinationsDetails';
+import ScheduleDetails from '../components/ScheduleDetails';
+import ReportJobStatus from './ReportJobStatus';
+import JobDetails from './JobDetails';
+
+export type RunHistoryProps = {
+    reportId: string;
+};
+
+function ReportJobs({ reportId }: RunHistoryProps) {
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const [filteredStatuses, setFilteredStatuses] = useState<string[]>([]);
+    const [showOnlyMyJobs, setShowOnlyMyJobs] = React.useState<boolean>(false);
+    const expandedRowSet = useSet<string>();
+
+    const { reportSnapshots, isLoading, error } = useFetchReportHistory({
+        id: reportId,
+        query: '',
+        page,
+        perPage,
+    });
+
+    const handleChange = (checked: boolean) => {
+        setShowOnlyMyJobs(checked);
+    };
+
+    if (isLoading) {
+        return (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <NotFoundMessage
+                title="Error fetching report history"
+                message={error || 'No data available'}
+            />
+        );
+    }
+
+    if (!reportSnapshots.length) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate title="No run history" headingLevel="h2" icon={CubesIcon} />
+            </Bullseye>
+        );
+    }
+
+    return (
+        <>
+            <Toolbar>
+                <ToolbarContent>
+                    <ToolbarItem>
+                        <CheckboxSelect
+                            ariaLabel="CVE severity checkbox select"
+                            toggleIcon={<FilterIcon />}
+                            selections={filteredStatuses}
+                            onChange={setFilteredStatuses}
+                            placeholderText="Filter by status"
+                        >
+                            <SelectOption value="PREPARING">Preparing</SelectOption>
+                            <SelectOption value="WAITING">Waiting</SelectOption>
+                            <SelectOption value="SUCCESS">Successful</SelectOption>
+                            <SelectOption value="ERROR">Error</SelectOption>
+                        </CheckboxSelect>
+                    </ToolbarItem>
+                    <ToolbarItem className="pf-u-flex-grow-1">
+                        <Switch
+                            id="view-only-my-jobs"
+                            label="View only my jobs"
+                            labelOff="View only my jobs"
```

Comment:
Would it make sense for this to read `View all jobs` when the switch is off or do you think the changing text would be confusing?

Reply:
Yea, I think, in my opinion, it might be confusing if the text changes. But might be different for others. @zhenpesky what do you think about the changing text idea?

Reply:
@dvail @sachaudh  I would like to keep it as "View only my jobs".
I thought about changing to other text and altering the text when switching on / off. 
However, I think in the context we have it here is more than just turning on and off a Bluetooth function, it makes sense to change the text when turning on and off WiFi or Bluetooth and it is not changing any existing things that affect the specific elements in the UI instead it affects the whole device. 
Take a closer look at our case, we are reflecting changes to a list of data that is already shown in both substances (including my jobs and others' jobs and those specific items in the list will be changing constantly).
The tab is already saying all report jobs. We would assume all report jobs would be displayed by default and thus there is no need to have a toggle as "view all jobs" again. And we don't want to change the tab's name to my report jobs because that doesn't capture our functionality in reviewing all report jobs when landing on this report page in general. 

Another reason for this is the consistency that https://console.redhat.com/openshift (feel free to take a look at the 'cluster' menu) is already using a pattern like this where they don't change the text and show all clusters by default.